From f66adc907c6b347e4ef6ddd9a6c77f9c84a39efe Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 28 Jun 2024 15:54:02 +0200
Subject: [PATCH 641/954] refactor(stm32mp2): remove NS_SYSRAM

The non-secure SYSRAM area was previously used for SCMI messages. This
is no more the case, OP-TEE manages a shared buffer in its memory.
Remove all mentions to STM32MP_NS_SYSRAM_* macros, and remove the
mapping of this area.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git cabf30038ee04e65c541e7e3c88c686382b44d67 refactor(stm32mp2): remove NS_SYSRAM]
Change-Id: I539dd90245a484bb1f5790d17eda4591e9f44d68
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/390447
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 17 ++---------------
 plat/st/stm32mp2/stm32mp2_private.c |  9 ---------
 2 files changed, 2 insertions(+), 24 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 3918126db0..7fbd07d8ad 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -149,8 +149,7 @@
 
 #define STM32MP_SEC_SYSRAM_BASE		STM32MP_SYSRAM_BASE
 
-#if defined(IMAGE_BL2)
-#if STM32MP_USB_PROGRAMMER && !STM32MP21
+#if defined(IMAGE_BL2) && STM32MP_USB_PROGRAMMER && !STM32MP21
 #define STM32MP_SEC_DEVICE_SYSRAM_SIZE		STM32MP_USB_DWC3_SIZE
 #define STM32MP_SEC_DEVICE_SYSRAM_BASE		(STM32MP_SYSRAM_BASE + \
 						 STM32MP_SYSRAM_SIZE - \
@@ -161,20 +160,8 @@
 
 #define STM32MP_USB_DWC3_BASE		STM32MP_SEC_DEVICE_SYSRAM_BASE
 #define STM32MP_USB_DWC3_SIZE		PAGE_SIZE
-#else /* STM32MP_USB_PROGRAMMER && !STM32MP21 */
+#else /* IMAGE_BL2 && STM32MP_USB_PROGRAMMER && !STM32MP21 */
 #define STM32MP_SEC_SYSRAM_SIZE		STM32MP_SYSRAM_SIZE
-#endif /* STM32MP_USB_PROGRAMMER && !STM32MP21 */
-#else /* IMAGE_BL2 */
-#define STM32MP_NS_SYSRAM_SIZE		PAGE_SIZE
-#define STM32MP_NS_SYSRAM_BASE		(STM32MP_SYSRAM_BASE + \
-					 STM32MP_SYSRAM_SIZE - \
-					 STM32MP_NS_SYSRAM_SIZE)
-
-#define STM32MP_SCMI_NS_SHM_BASE	STM32MP_NS_SYSRAM_BASE
-#define STM32MP_SCMI_NS_SHM_SIZE	STM32MP_NS_SYSRAM_SIZE
-
-#define STM32MP_SEC_SYSRAM_SIZE		(STM32MP_SYSRAM_SIZE - \
-					 STM32MP_NS_SYSRAM_SIZE)
 #endif
 
 /* DDR configuration */
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index b5e9f053a3..5f0a0e2873 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -48,14 +48,6 @@
 					MT_EXECUTE_NEVER)
 #endif /* STM32MP_USB_PROGRAMMER && !STM32MP21 */
 
-/* Non-secure SYSRAM is used a uncached memory for SCMI message transfer */
-#define MAP_NS_SYSRAM	MAP_REGION_FLAT(STM32MP_NS_SYSRAM_BASE, \
-					STM32MP_NS_SYSRAM_SIZE, \
-					MT_DEVICE | \
-					MT_RW | \
-					MT_NS | \
-					MT_EXECUTE_NEVER)
-
 #define MAP_SRAM1	MAP_REGION_FLAT(SRAM1_BASE, \
 					SRAM1_MAP_SIZE, \
 					MT_MEMORY | \
@@ -86,7 +78,6 @@ static const mmap_region_t stm32mp2_mmap[] = {
 #if defined(IMAGE_BL31)
 static const mmap_region_t stm32mp2_mmap[] = {
 	MAP_SEC_SYSRAM,
-	MAP_NS_SYSRAM,
 	MAP_SRAM1,
 	MAP_DEVICE,
 	{0}
-- 
2.53.0

